lib/python/rose/bush.py

```diff
@@ -517,7 +519,48 @@ def view(self, user, suite, path, path_in_tar=None, mode=None):
             file_content = "rose-conf"
         else:
             file_content = self.suite_engine_proc.is_conf(path)
-        template = self.template_env.get_template("view.html")
+
+        line_numbers = None # used to return line numbers matching a search
+        template = None
+
+        if search_string == "":
+            # no search is being performed, client is refresing the file
+            template = self.template_env.get_template("search.html")
+        elif search_string:
+            # client is performing a search
+            template = self.template_env.get_template("search.html")
+            results = []
+            line_numbers = []
+            mark = ('<span class="highlight">', '</span>') # mark matches
```

Comment:
The problem of this is that we are leaking HTML strings from Jinja2 templates into Python.